### PR TITLE
SOC-2445 tests for: delete all posts by user

### PR DESCRIPTION
**Pull request by tortila**
Ticket: https://wikia-inc.atlassian.net/browse/SOC-2445
Test scenarios described: https://wikia-inc.atlassian.net/wiki/display/SOC/Deleting+all+posts+by+user

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteDialog.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.common;
+
+
+import lombok.AllArgsConstructor;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteDialog.java]: MINOR Remove this unused import 'lombok.AllArgsConstructor'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/common/templates/NewTestTemplate.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import java.lang.reflect.Method;
 
+import org.apache.http.client.methods.HttpRequestBase;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/templates/NewTestTemplate.java]: MINOR Remove this unused import 'org.apache.http.client.methods.HttpRequestBase'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/common/remote/operations/http/BaseRemoteOperation.java [hunk 3]
```diff
@@ -62,7 +62,8 @@ private String makeRequest(final CloseableHttpClient client, final HttpRequestBa
     String result = StringUtils.EMPTY;
 
     try {
-      request.setHeader(Discussions.ACCESS_TOKEN_HEADER, Helios.getAccessToken(user));
+      if(user != null)
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/http/BaseRemoteOperation.java]: MAJOR Missing curly brace. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00121)

File: src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package com.wikia.webdriver.common.remote.operations.http;
+
+
+import com.wikia.webdriver.common.core.Helios;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.Helios'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import com.wikia.webdriver.common.core.helpers.User;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.helpers.User'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+import com.wikia.webdriver.common.remote.Discussions;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.remote.Discussions'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java]: MINOR Remove this unused import 'org.apache.http.client.methods.HttpPost'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.json.JSONObject;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java]: MINOR Remove this unused import 'org.json.JSONObject'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+import static com.wikia.webdriver.elements.mercury.components.discussions.common.DiscussionsConstants.DESKTOP_RESOLUTION;
+
+
+/**
+ * Tests for deleting all posts by some user
+ */
+@Execute(onWikia = MercuryWikis.DISCUSSIONS_AUTO)
+public class DeleteAllPostsByUserTests extends NewTestTemplate {
+
+  private User UserWithPosts = User.USER_6;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java]: MAJOR Rename this field "UserWithPosts" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)

File: src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+  private String siteId;
+
+  /**
+   * Hacky way to get rid of totally redundant loading of Special:Version.
+   * The only reason it was loaded was to extract siteId; in this case siteId is extracted from
+   * response of the same page, but parsing is via regex, once, so no need to load the page each time
+   */
+  //@Override
+  //protected void loadFirstPage() {}
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s1/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+
+import com.wikia.webdriver.common.contentpatterns.MercuryWikis;
+import com.wikia.webdriver.common.contentpatterns.URLsContent;
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.core.annotations.Execute;
+import com.wikia.webdriver.common.core.annotations.InBrowser;
+import com.wikia.webdriver.common.core.drivers.Browser;
+import com.wikia.webdriver.common.core.helpers.Emulator;
+import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.common.core.imageutilities.Shooter;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.imageutilities.Shooter'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+  private String wikiUrl;
+
+  private void setSiteId() {
+    NoAuthOperation request = new NoAuthOperation();
+    String response = request.execute(this.wikiUrl + URLsContent.SPECIAL_VERSION, new JSONObject());
+    Pattern p = Pattern.compile(".*city_id: (\\d+).*", Pattern.DOTALL);
+    Matcher m = p.matcher(response);
+    if (m.find())
+      this.siteId = m.group(1);
+    else
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java]: MAJOR Missing curly brace. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00121)

File: src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+
+public class GetSiteId {
+
+  public GetSiteId(String wikiUrl) {
+    this.wikiUrl = wikiUrl;
+    this.setSiteId();
+  }
+
+  @Getter
+  private String siteId;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+  private String wikiUrl;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java [hunk 15]
```diff
@@ -0,0 +1,35 @@
+package com.wikia.webdriver.common.remote.operations;
+
+
+import com.wikia.webdriver.common.contentpatterns.URLsContent;
+import com.wikia.webdriver.common.remote.operations.http.NoAuthOperation;
+import lombok.Getter;
+import org.apache.http.client.methods.HttpGet;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java]: MINOR Remove this unused import 'org.apache.http.client.methods.HttpGet'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+import org.apache.http.client.methods.HttpPost;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java]: MINOR Remove this unused import 'org.apache.http.client.methods.HttpPost'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+  @Getter
+  private String siteId;
+  private String wikiUrl;
+
+  private void setSiteId() {
+    NoAuthOperation request = new NoAuthOperation();
+    String response = request.execute(this.wikiUrl + URLsContent.SPECIAL_VERSION, new JSONObject());
+    Pattern p = Pattern.compile(".*city_id: (\\d+).*", Pattern.DOTALL);
+    Matcher m = p.matcher(response);
+    if (m.find())
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java]: MAJOR Missing curly brace. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00121)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+  private WebElement button;
+
+  // context of DeleteAllButton in which it should be displayed
+  @FindBy(css = ".discussion-page.row > div:nth-child(3).rail")
+  private WebElement rightRail;
+
+  public boolean isVisible() {
+    try {
+      return button.isDisplayed();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+      return false;
+    }
+  }
+
+  public boolean isNotVisible() {
+    try {
+      return rightRail.isEnabled() && wait.forElementNotVisible(button);
+    } catch (TimeoutException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 19 issues:
* MAJOR 7 major
* MINOR 12 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/common/remote/operations/DiscussionsOperations.java [hunk 20]
```diff
@@ -17,8 +17,14 @@
 
   private final WebDriver driver;
 
-  public PostEntity.Data cratePostWithUniqueData() {
-    return createPost(CreatePostContext.defaultContext(extractSiteId()));
+  // callable only when on Special:Version
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/common/remote/operations/DiscussionsOperations.java]: Can we use Javadoc here?
```
/**
 * Callable only when on Special:Version
 *
**/
```

File: src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+import com.wikia.webdriver.common.remote.operations.http.NoAuthOperation;
+import lombok.Getter;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.json.JSONObject;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class GetSiteId {
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java]: Can this have a different name?

File: src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java [hunk 22]
```diff
@@ -0,0 +1,35 @@
...
+  public GetSiteId(String wikiUrl) {
+    this.wikiUrl = wikiUrl;
+    this.setSiteId();
+  }
+
+  @Getter
+  private String siteId;
+  private String wikiUrl;
+
+  private void setSiteId() {
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/common/remote/operations/GetSiteId.java]: `extractSiteIdFromWikiUrl`?

File: src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java [hunk 23]
```diff
@@ -0,0 +1,37 @@
...
+public class NoAuthOperation extends PostRemoteOperation {
+  public NoAuthOperation() {
+    super(null);
+  }
+
+  private String makeRequest(final CloseableHttpClient client, final HttpRequestBase request)
+    throws IOException {
+    String result = StringUtils.EMPTY;
+    try {
+      try (CloseableHttpResponse response = client.execute(request)) {
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/common/remote/operations/http/NoAuthOperation.java]: I believe you can combine the try-with-resources with try-catch.

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/ConfirmationDialog.java [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+      // allow selenium to throw exception
+      dialog = driver.findElement(By.cssSelector(".discussion-dialog.is-visible.modal-dialog-approve"));
+    } else if (1 < elements.size()) {
+      // only one modal dialog should be visible
+      throw new IllegalStateException("Only one modal dialog should be visible!");
+    } else {
+      dialog = elements.get(0);
+    }
+
+    return dialog;
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/ConfirmationDialog.java]: Remove the pre-defined variable?
```
   private WebElement getActiveDialog() {
     List<WebElement> elements = driver.findElements(By.cssSelector(".discussion-dialog.is-visible"));
     if (elements.isEmpty()) {
       // allow selenium to throw exception
       return driver.findElement(By.cssSelector(".discussion-dialog.is-visible.modal-dialog-approve"));
     } else if (1 < elements.size()) {
       // only one modal dialog should be visible
       throw new IllegalStateException("Only one modal dialog should be visible!");
     } else {
       return elements.get(0);
     }
```

File: src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java [hunk 25]
```diff
@@ -0,0 +1,247 @@
...
+  }
+
+  // GENERAL - no specific user permissions
+
+  @Test(groups = "discussions-deleteAllPostsByUser")
+  @Execute(asUser = User.STAFF)
+  @InBrowser(browserSize = DESKTOP_RESOLUTION)
+  public void clickDeleteAllDesktopDisplaysConfirmationDialog() {
+    DeleteDialog confirmationDialog = getDeleteAllButton(UserWithPosts.getUserId()).click();
+    Assertion.assertTrue(confirmationDialog.isVisible());
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java]: Static import the `assertTrue`?

**Comment by WikiaTech** [comment]
SonarQube analysis reported 19 issues:
* MAJOR 7 major
* MINOR 12 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java [hunk 26]
```diff
@@ -0,0 +1,35 @@
...
+
+public class SiteId {
+
+  public SiteId(String wikiUrl) {
+    this.wikiUrl = wikiUrl;
+    this.extractSiteIdFromSpecialVersion();
+  }
+
+  @Getter
+  private String siteId;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java [hunk 27]
```diff
@@ -0,0 +1,35 @@
...
+  private String wikiUrl;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java [hunk 28]
```diff
@@ -0,0 +1,35 @@
...
+
+  private void extractSiteIdFromSpecialVersion() {
+    NoAuthOperation request = new NoAuthOperation();
+    String response = request.execute(this.wikiUrl + URLsContent.SPECIAL_VERSION, new JSONObject());
+    Pattern p = Pattern.compile(".*city_id: (\\d+).*", Pattern.DOTALL);
+    Matcher m = p.matcher(response);
+    if (m.find()) {
+      this.siteId = m.group(1);
+    }
+    else {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java]: MAJOR Move this "else" on the same line that the previous closing curly brace. [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck)

File: src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java [hunk 29]
```diff
@@ -0,0 +1,239 @@
...
+import static com.wikia.webdriver.common.core.Assertion.assertFalse;
+
+
+/**
+ * Tests for deleting all posts by some user
+ */
+@Execute(onWikia = MercuryWikis.DISCUSSIONS_AUTO)
+public class DeleteAllPostsByUserTests extends NewTestTemplate {
+
+  private final User USER_WITH_POSTS = User.USER_6;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java]: MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)
MAJOR Rename this field "USER_WITH_POSTS" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java [hunk 30]
```diff
@@ -0,0 +1,42 @@
...
+      System.err.println(e.toString());
+      return false;
+    }
+  }
+
+  public boolean isNotVisible() {
+    try {
+      return rightRail.isEnabled() && wait.forElementNotVisible(button);
+    } catch (TimeoutException e) {
+      System.err.println(e.toString());
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS106)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java [hunk 31]
```diff
@@ -0,0 +1,42 @@
...
+
+  // context of DeleteAllButton in which it should be displayed
+  @FindBy(css = ".discussion-page.row > div:nth-child(3).rail")
+  private WebElement rightRail;
+
+  public boolean isVisible() {
+    try {
+      return button.isDisplayed();
+    } catch (NoSuchElementException e) {
+      System.err.println(e.toString());
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS106)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues:
* MAJOR 7 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by slayful** [comment]
LGTM

File: src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java (hunk 26 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java (hunk 27 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java (hunk 28 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/SiteId.java]: MAJOR Move this "else" on the same line that the previous closing curly brace. [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck)

File: src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java (hunk 29 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java]: MAJOR Rename this field "USER_WITH_POSTS" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)
MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java (hunk 30 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS106)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java [hunk 32]
```diff
@@ -0,0 +1,42 @@
...
+  private WebElement button;
+
+  // context of DeleteAllButton in which it should be displayed
+  @FindBy(css = ".discussion-page.row > div:nth-child(3).rail")
+  private WebElement rightRail;
+
+  public boolean isVisible() {
+    try {
+      return button.isDisplayed();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java (hunk 31 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS106)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java [hunk 33]
```diff
@@ -0,0 +1,42 @@
...
+      System.err.println(e.toString());
+      return false;
+    }
+  }
+
+  public boolean isNotVisible() {
+    try {
+      return rightRail.isEnabled() && wait.forElementNotVisible(button);
+    } catch (TimeoutException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues:
* MAJOR 7 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java [hunk 34]
```diff
@@ -0,0 +1,44 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.common;
+
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+import mx4j.log.Logger;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java]: MINOR Remove this unused import 'mx4j.log.Logger'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Pull request merged**